File: cookbook/copilot-sdk/go/recipe/error-handling.go
package main

import (
	"context"
	"fmt"
	"log"

	copilot "github.com/github/copilot-sdk/go"
)

func main() {
	ctx := context.Background()
	client := copilot.NewClient(nil)

	if err := client.Start(ctx); err != nil {
		log.Fatalf("無法啟動用戶端: %v", err)
	}
	defer client.Stop()

	session, err := client.CreateSession(ctx, &copilot.SessionConfig{
		OnPermissionRequest: copilot.PermissionHandler.ApproveAll,
		Model:               "gpt-5.4",
	})
	if err != nil {
		log.Fatalf("建立工作階段失敗: %v", err)
	}
	defer session.Disconnect()

	result, err := session.SendAndWait(ctx, copilot.MessageOptions{Prompt: "Hello!"})
	if err != nil {
		log.Printf("傳送訊息失敗: %v", err)
		return
	}

	if result != nil {
		if d, ok := result.Data.(*copilot.AssistantMessageData); ok {
			fmt.Println(d.Content)
		}
	}
}

File: cookbook/copilot-sdk/dotnet/recipe/error-handling.cs
#:package GitHub.Copilot.SDK@*
#:property PublishAot=false

// GitHub.Copilot.SDK 套件公開了 GitHub.Copilot 命名空間。
using GitHub.Copilot;

var client = new CopilotClient();

try
{
    await client.StartAsync();
    var session = await client.CreateSessionAsync(new SessionConfig
    {
        Model = "gpt-5",
        OnPermissionRequest = PermissionHandler.ApproveAll
    });

    var done = new TaskCompletionSource<string>();
    session.On(evt =>
    {
        if (evt is AssistantMessageEvent msg)
        {
            done.SetResult(msg.Data.Content);
        }
    });

    await session.SendAsync(new MessageOptions { Prompt = "Hello!" });
    var response = await done.Task;
    Console.WriteLine(response);

    await session.DisposeAsync();
}
catch (Exception ex)
{
    Console.WriteLine($"錯誤: {ex.Message}");
}
finally
{
    await client.StopAsync();
}
